### ISSUE TITLE
Site is very slow

### ISSUE THREAD

**Issue by nithiinl** [ThilakReddyy/JNTUH_RESULTS#1]
Hey bro the website u developed is very useful for b
Students who r studying in jntuh. All of my friends are using ur website to download results. But the thing is, the website is very slow, if any possibilities are there to fix this please do it bro. It's very useful for us

**Comment by ThilakReddyy** [comment]
> Hey bro the website u developed is very useful for b Students who r studying in jntuh. All of my friends are using ur website to download results. But the thing is, the website is very slow, if any possibilities are there to fix this please do it bro. It's very useful for us

I am an undergrad student same as you are. As the heroku is free hosting platform I have deployed my app. If you want to contribute you are welcome.Anyways you can use the alternative website
Alternative website: jntuhresult.herokuapp.com

**Comment by Syed-Ansar** [comment]
Hey! @ThilakReddyy I am Syed Ansar. This is very helpful for JNTUH Students. I would like to contribute. We can make this fast or can improve its performance by using React or Nextjs.

**Comment by ThilakReddyy** [comment]
You can raise a Pull Request and make your changes. If you do so try to make the script more generic.

**Comment by Syed-Ansar** [comment]
Yes! I wish to do but I am not aware of python/django. If there is an api then I can make good interface and performance.

**Comment by hemanth-kotagiri** [comment]
Hello there, @Syed-Ansar and @ThilakReddyy . I have come across your website just now, and its a game changer.
I would invite you both to collaborate with me on the following projects:
I have created an entire suite of JNTUH results applications as well, you can find them here:

Website built with Next.js
https://jntuh-results-stats.vercel.app/

Backend API
[results-restapi.herokuapp.com/](https://results-restapi.herokuapp.com/)
https://github.com/hemanth-kotagiri/sgpa-rest-api

Documentation for the API
https://hemanthk.me/sgpa-rest-api-docs/

Mobile Application built with Flutter
https://github.com/hemanth-kotagiri/sgpa-rest-api

I would please you both to join the organization I created here:
https://github.com/jntuh-results-services

@ThilakReddyy, I just had a glance at your code for this application. The reason it is being slow is that it is not asynchronous. You are trying to fetch each student details/results synchronously - where each request is independent of each other. You can simply shoot multiple async requests to JNTUH servers like I have done for getting results of more than 170 halltickets in less than 2 seconds. Let us have a conversation and implement that into your application if you're willing to. 

Looking forward to collab! Cheers!

**Comment by Syed-Ansar** [comment]
Hello @hemanth-kotagiri It is a great idea. Let's have a chat.

**Comment by ThilakReddyy** [comment]
Sure. Could you connect with me on LinkedIn https://www.linkedin.com/in/thilakreddyy

**Comment by Mohankumar0521** [comment]
This site is not working
